show-magnifier/src/main/java/com/esri/arcgismaps/sample/showmagnifier/components/MapViewModel.kt

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.showmagnifier.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.InteractionOptions
+import com.arcgismaps.mapping.view.MapViewInteractionOptions
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+
+class MapViewModel(application: Application) : AndroidViewModel(application) {
+    // set the MapView mutable stateflow
+    val mapViewState = MutableStateFlow(MapViewState())
+}
+
+
+/**
+ * Data class that represents the MapView state
+ */
+data class MapViewState( // This would change based on each sample implementation
```

Comment:
This is just a stub comment, not required in the sample 😅 

```suggestion
data class MapViewState( 
```